server/backbone/api/helpers.py

```diff
@@ -0,0 +1,95 @@
+import itertools
+from datetime import timedelta, datetime
+
+from flask_login import current_user
+from sqlalchemy import inspect
+
+from ..exceptions import BackboneException
+from ..models import Parent, Child
+
+ALLOWED_EXTENSIONS = ['jpg', 'jpeg', 'png', 'gif']
+
+
+def generate_prnt_resp(prnt):
+    """
+    Generate a dictionary description of a Parent object, used by for example /me
+    :param prnt: a Parent object
+    :return: a dictionary of description of Parent object
+    """
+    d = {}
+    for e in ['email', 'name', 'id']:
+        d[e] = getattr(prnt, e)
+    d['children'] = [c.id for c in prnt.children]
+    return d
+
+
+def generate_chd_resp(chd):
+    """
+    Generate a dictionary description of a Child object, used by for example /me
+    :param chd: a Child object
+    :return: a dictionary of description of Child object
+    """
+    d = {}
+    for e in ['level', 'name', 'id', 'xp']:
+        d[e] = getattr(chd, e)
+    return d
+
+
+def generate_qst_resp(qst, ts=datetime.utcnow()):
+    """
+    Generate a dictionary description of a Quest object, used by for example /get_quest
+    :param qst: a Quest object
+    :param ts: a datetime object that we want to get next occurrence from
+    :return: a dictionary of description of Quest object
+    """
+    d = {}
+    for e in ['title', 'description', 'reward', 'due', 'id', 'recurring']:
+        d[e] = getattr(qst, e)
+    # Add extra values that we don't store in database
+    looking_for = find_next_time(qst, ts) if qst.recurring else qst.due
+    d['due'] = datetime.timestamp(d['due'])
+    if qst.recurring:
+        d['next_occurrence'] = datetime.timestamp(looking_for)
+    filtered_completions = list(filter(lambda c: c.value == looking_for, qst.completions))
+    d['completed_on'] = filtered_completions[0].ts if len(filtered_completions) == 1 else None
+    return d
+
+
+def me_or_my_child(child_obj):
```

Comment:
either split this into two separate methods: "owns_child" and "is_child"

or rename this method to "owns_child_or_is_child"

Reply:
To me `is_child` means something else than what this function's half does.